network/src/protocols/mod.rs

```diff
@@ -268,12 +268,6 @@ impl ServiceProtocol for CKBHandler {
     }
 
     fn received(&mut self, context: ProtocolContextMutRef, data: Bytes) {
-        self.network_state.with_peer_registry_mut(|reg| {
-            if let Some(peer) = reg.get_peer_mut(context.session.id) {
-                peer.last_message_time = Some(Instant::now());
```

Comment:
Technically, this field represents the time of the last message we received, not just for the ping message. So I am not sure if you can safely remove it.

Reply:
Instead of removing here, actually, the updating on the Pong message is redundant.

Reply:
The only effect of this value is to serve as one of the reference conditions for eviction when the upper limit of the connection is reached. In other words, we do not need to update it so frequently, the cost is too high and the benefit is low

Reply:
The problem is you break the feature of this field:

1. the meaning of this field is `last_message_time`, but the implementation is a different thing if another piece of code relies on this field they may get an unexpected result.
2. The old eviction will consider all messages time, now we only consider the last ping time, a ping message is sent every 15 seconds, now an honest node will get higher probabilities to get evicted during the gap, even the node relyed more transactions or blocks than others.

I am doubtful about the performance benefit trade-off here, these two lines try to acquire a lock then update a field, itself is a very fast operation. If you find some performance bottleneck in here via some measurement, which means some other pieces of code holds the lock, I think the better solution is to digger into that lock and reduce the lock acquiring time.

Reply:
Any protocol implemented based on the handle will be forced to update this field. The more protocols, the more severe the conflict, that is, the processing message is unilaterally changed to a blocked state

Reply:
Rename it to `last_ping_time`?